InternetHospital.BusinessLogic/Services/DoctorService.cs

```diff
@@ -0,0 +1,47 @@
+﻿using InternetHospital.BusinessLogic.Interfaces;
+using InternetHospital.BusinessLogic.Models;
+using InternetHospital.DataAccess;
+using System.Linq;
+
+namespace InternetHospital.BusinessLogic.Services
+{
+    public class DoctorService : IDoctorService
+    {
+        private ApplicationContext _context;
+
+        public DoctorService(ApplicationContext context)
+        {
+            _context = context;
+        }
+
+        public (IQueryable<DoctorsDto> doctors, int count) GetAll(DoctorSearchParameters queryParameters)
+        {
+            var _doctors = _context.Doctors.AsQueryable();
+            if (queryParameters.SearchByName != null)
+            {
+                _doctors = _doctors
+                    .Where(x => x.User.FirstName.ToLower().Contains(queryParameters.SearchByName.ToLower())
+                    || x.User.SecondName.ToLower().Contains(queryParameters.SearchByName.ToLower())
+                    || x.User.ThirdName.ToLower().Contains(queryParameters.SearchByName.ToLower()));
```

Comment:
It would be better just to replace value by it's lowercase value. 
queryParametrs.SearchByName = queryParametrs. SearchByName.ToLower();